SQL: Cast on generated -NULL value

This causes an error value in sequential runtime.

```
$ super -version
Version: 5ea0cb5d6

$ super -c "SELECT CAST(-CASE WHEN 1=0 THEN 1 ELSE NULL END AS INTEGER) AS col0;"
{col0:error({message:"type incompatible with unary '-' operator",on:null})}
```

And a panic in vector runtime.

```
$ SUPER_VAM=1 super -c "SELECT CAST(-CASE WHEN 1=0 THEN 1 ELSE NULL END AS INTEGER) AS col0;"
panic: &{Typ:0xc0004c9bc0 Vals:0xc00017c000 Nulls:empty}
goroutine 1 [running]:
runtime/debug.Stack()
	/usr/local/opt/go/libexec/src/runtime/debug/stack.go:26 +0x5e
github.com/brimdata/super/runtime/sam/op.(*Catcher).Pull.func1()
	/Users/phil/work/super/runtime/sam/op/catcher.go:25 +0x3d
panic({0x36f4d60?, 0xc0004d3900?})
	/usr/local/opt/go/libexec/src/runtime/panic.go:783 +0x132
github.com/brimdata/super/runtime/vam/expr/cast.toNumeric[...]({0x3942970?, 0xc0004d3900?}, {0x3937318?, 0x4b1a880?}, {0x0?, 0x0?, 0xc0001171e0?})
	/Users/phil/work/super/runtime/vam/expr/cast/number.go:85 +0x4d6
github.com/brimdata/super/runtime/vam/expr/cast.castToNumber({0x3942970, 0xc0004d3900}, {0x3937318, 0x4b1a880}, {0x0, 0x0, 0x0})
	/Users/phil/work/super/runtime/vam/expr/cast/number.go:31 +0x3d5
github.com/brimdata/super/runtime/vam/expr/cast.To.func1({0x3942970?, 0xc0004d3900?}, {0x0?, 0xc00017c000?, 0xc0004c9bc0?})
	/Users/phil/work/super/runtime/vam/expr/cast/cast.go:17 +0x39
github.com/brimdata/super/runtime/vam/expr/cast.assemble(0xc000134360, {0x3942970, 0xc0004d3900}, {0x3937318, 0x4b1a880}, 0x1d?)
	/Users/phil/work/super/runtime/vam/expr/cast/cast.go:66 +0x44c
github.com/brimdata/super/runtime/vam/expr/cast.To(0xc000134360, {0x3942970?, 0xc0004d3900?}, {0x3937318, 0x4b1a880})
	/Users/phil/work/super/runtime/vam/expr/cast/cast.go:39 +0x17c
github.com/brimdata/super/runtime/vam/expr.(*casterPrimitive).Eval(0x0?, {0x3942970?, 0xc0004d3900?})
	/Users/phil/work/super/runtime/vam/expr/cast.go:54 +0x2b
github.com/brimdata/super/runtime/vam/expr.(*literalCast).Eval.func1({0xc0004fda20?, 0x10?, 0xc000680008?})
	/Users/phil/work/super/runtime/vam/expr/cast.go:44 +0x33
github.com/brimdata/super/vector.Apply(0x48?, 0x3942910?, {0xc0004fda20?, 0x0?, 0x3942620?})
	/Users/phil/work/super/vector/apply.go:19 +0xf3
github.com/brimdata/super/runtime/vam/expr.(*literalCast).Eval(0xc0001268a0, {0x3942910?, 0xc0004b5cc0?})
	/Users/phil/work/super/runtime/vam/expr/cast.go:43 +0x9e
github.com/brimdata/super/runtime/vam/expr.(*recordExpr).Eval(0xc000136850, {0x3942910, 0xc0004b5cc0})
	/Users/phil/work/super/runtime/vam/expr/recordexpr.go:39 +0x1a2
github.com/brimdata/super/runtime/vam/op.(*Values).Pull(0xc000119620, 0x0)
	/Users/phil/work/super/runtime/vam/op/values.go:34 +0x10c
github.com/brimdata/super/runtime/vam.(*Materializer).Pull(0xc00055fa18?, 0x50?)
	/Users/phil/work/super/runtime/vam/materialize.go:26 +0x27
github.com/brimdata/super/runtime/sam/op.(*Single).Pull(0xc000119680, 0x10?)
	/Users/phil/work/super/runtime/sam/op/mux.go:120 +0x33
github.com/brimdata/super/runtime/sam/op.(*Catcher).Pull(0x6c4831d599e6c44f?, 0x25?)
	/Users/phil/work/super/runtime/sam/op/catcher.go:28 +0x5c
github.com/brimdata/super/runtime/exec.(*Query).Pull(0x1910a3f?, 0x0?)
	/Users/phil/work/super/runtime/exec/query.go:49 +0x3c
github.com/brimdata/super/sbuf.CopyMux(0xc00055fd10, {0x392ec80, 0xc0001196b0})
	/Users/phil/work/super/sbuf/mux.go:39 +0x38
github.com/brimdata/super/cmd/super/root.(*Command).Run(0xc0004066c8, {0xc00003e8f0, 0x0, 0x0})
	/Users/phil/work/super/cmd/super/root/command.go:167 +0xa19
github.com/brimdata/super/pkg/charm.path.run({0xc00008aa70, 0x1, 0x1}, {0xc00003e8f0, 0x0, 0x0?})
	/Users/phil/work/super/pkg/charm/path.go:11 +0x7b
github.com/brimdata/super/pkg/charm.(*Spec).Exec(0x4adf160, {0xc00003e8e0, 0x2, 0x2})
	/Users/phil/work/super/pkg/charm/charm.go:74 +0x1fa
main.main()
	/Users/phil/work/super/cmd/super/main.go:39 +0x5b
```

## Details

Repro is with super commit 5ea0cb5. The query above is a simplification of the one from sqllogictest [random/expr/slt_good_100/q13935](https://github.com/brimdata/sqllogic-ztests/blob/main/ztests/sqlite/random/expr/slt_good_100/ztests-1/q13935.yaml).

Here it is returning the expected result in Postgres.

```
$ psql postgres
psql (17.7 (Homebrew))
Type "help" for help.

postgres=# \pset null 'NULL'
Null display is "NULL".
postgres=# SELECT CAST(-CASE WHEN 1=0 THEN 1 ELSE NULL END AS INTEGER) AS col0;
 col0 
------
 NULL
(1 row)
```

--- a/runtime/sam/expr/eval.go
+++ b/runtime/sam/expr/eval.go
@@ -581,7 +581,7 @@ func (u *UnaryMinus) Eval(this super.Value) super.Value {
 		}
 		val = super.NewInt(typ, v)
 	}
-	if val.IsNull() && super.IsNumber(typ.ID()) {
+	if val.IsNull() && (super.IsNumber(typ.ID()) || typ.ID() == super.IDNull) {
 		return val
 	}
 	switch typ.ID() {

--- a/runtime/vam/expr/unaryminus.go
+++ b/runtime/vam/expr/unaryminus.go
@@ -31,7 +31,10 @@ func (u *unaryMinus) eval(vecs ...vector.Any) vector.Any {
 		return vec
 	}
 	id := vec.Type().ID()
-	if !super.IsNumber(vec.Type().ID()) {
+	if id == super.IDNull {
+		return vec
+	}
+	if !super.IsNumber(id) {
 		return vector.NewWrappedError(u.sctx, "type incompatible with unary '-' operator", vecs[0])
 	}
 	if super.IsUnsigned(id) {

--- a/runtime/ztests/expr/unary-minus.yaml
+++ b/runtime/ztests/expr/unary-minus.yaml
@@ -62,7 +62,7 @@ output: |
   error({message:"type incompatible with unary '-' operator",on:"foo"})
   error({message:"type incompatible with unary '-' operator",on:10.0.0.1})
   error({message:"type incompatible with unary '-' operator",on:[1,2,3]})
-  error({message:"type incompatible with unary '-' operator",on:null})
+  null
   error({message:"type incompatible with unary '-' operator",on:null::string})
   null::int8
   null::int16

